### DISCUSSION TITLE
[Docs] Doctrine AsEntityListener doesn't work on cascade

### DISCUSSION THREAD

**Discussion by ToshY** [symfony/symfony#51537] [Q&A]
I've just started using ["Doctrine Entity Listeners"](https://symfony.com/doc/current/doctrine/events.html#doctrine-entity-listeners) and I've come to the realisation that if you define an entity listener it will only listen to explicit events on the specified entity. Thus if you define a relation to cascade persist/remove the child, the entity listener for the child will _not_ be triggered.

Example:

`ParentEntity.php`
```php
class ParentEntity
{
    #[ORM\OneToMany(targetEntity: ChildEntity::class, cascade: ['persist', 'remove'], orphanRemoval: true)]
    private Collection $childEntityCollection;

    public function __construct()
    {
        $this->childEntityCollection = new ArrayCollection();
    }

    public function getChildEntityCollection(): Collection
    {
        return $this->childEntityCollection;
    }
}
```

`ChildEntityRemovedListener.php`
```php
#[AsEntityListener(event: Events::postRemove, method: Events::postRemove, entity: ChildEntity::class)]
final class ChildEntityRemovedListener
{
    public function postRemove(ChildEntity $childEntity, PostRemoveEventArgs $event): void
    {
        dd('Im not here');
    }
}
````

Now the actual removing of the parent/children:

```php
# Does NOT trigger `ChildEntityRemovedListener`
$entityManager->remove($parentEntity);
$entityManager->flush();

# Does trigger `ChildEntityRemovedListener`
foreach ($parentEntity->getChildEntityCollection() as $childEntity) {
    $entityManager->remove($childEntity);
}
$entityManager->flush();
```

While this may make sense for those more experienced with doctrine events, this is confusing for someone new with these listeners. 
If this is indeed how the doctrine entity listeners are supposed to work, maybe it would be a good idea to add an additional note to the documentation explicitly stating that using cascade will not trigger the event for the child listener?

---

`doctrine/orm:2.15.5`

**Comment by cavasinf** [comment]
The Symfony documentation does not say anything about this. But in this case it uses "Doctrine events".

You can read the [Doctrine docs about events](https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/events.html), especially the [postUpdate, postRemove, postPersist](https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/events.html#postupdate-postremove-postpersist) for your part.

**Reply by ToshY** [reply]
Hey @cavasinf 👋 

I'm aware it uses Doctrine events, but as it was my first time using them, it was slightly confusing that an entity listener configured for a child does not get triggered when a cascade operation is specified on it's parent. Apparently you need to explictely perform the persist/remove operations on the child trigger the child listener.

Because this was somewhat puzzling for me, I was wondering if more people struggled with this, and if so maybe we could add a note to the documentation explictely stating the above.
